docs/add icon set

diba1013 opened the pull request:
# Description

This describes the steps to perform to add the your own `.svg` icons that can be referenced within `mkdocs.yml` as `theme.icon`.

A rendered version can be found [here](https://diba1013.github.io/mkdocs-material-playground/customization/icons/) (small changes done within this PR, but gives an overall overview) 

## State of this PR

I have built the documentation and verified the rendered result. As per discussion on #1585, I specified `docs/tutorials` as base branch (which is slightly outdated, thus this PR contains a half broken link to `../extensions/pymdown.md#icons`).

### Questions

1. How do you want this to be setup? For now I have gone for a style: "All necessary steps are already described in the main documentation, so this will only guide you through without explaining them in detail". Thus I have only linked to the more detailed references.
2. How do you do your links? For now I have used inline-style relative markdown links.

### mkdocs-material-extensions

I have tested this with [mkdocs-material-extensions](https://github.com/facelessuser/mkdocs-material-extensions) (since it is now referenced on the production site) and it does not pick up on your custom icon sets (a setup can be found [here](https://github.com/diba1013/mkdocs-material-playground/blob/master/mkdocs.yml), note the `:bootstrap-circle-square:` within the bug notice). @facelessuser I know [this was not intended in the first place](https://github.com/squidfunk/mkdocs-material/issues/1498#issuecomment-597678171) and I am happy to file an issue for that. I understand that it is a [pre-release](https://github.com/facelessuser/mkdocs-material-extensions/issues/2) and will be shipped with mkdocs-material eventually.

If this is intentional or not possible, I am happy to remove this notice.

facelessuser commented:
@diba1013 I will need you to explain the "bug". Are you referring to using icon syntax `:myicon:` in the yml? If so, that syntax only works in Markdown content as intended.

facelessuser commented:
I would remove the `bug` note as this is not currently a bug, but an enhancement. It is a beta, and as such, has only implemented official icon support. This was known at the time of the beta's creation.

facelessuser commented:
It may be reasonable to have a informative note, but I would not advertise it as a bug.

diba1013 commented:
> It may be reasonable to have an informative note, but I would not advertise it as a bug.

Thanks for the feedback. I changed this to an information notice and improved the wording.  Please feel free to improve upon it as you see fit.

squidfunk commented on docs/tutorials/icons.md:
```diff
@@ -0,0 +1,38 @@
+
+## Usage
+
+You can use the icon within `mkdocs.yml`. For the example below, I assume the namespace `bootstrap` from [bootstrap-icons](https://icons.getbootstrap.com/).
```
Please don't use first person singular in the documentation.

squidfunk commented:
Thanks for the writeup! Regarding your question to which degree the tutorial should describe stuff that is already outlined in other parts of the documentation I'd recommend to maybe go in some more detail. A tutorial/howto should be understandable without having to navigate around a lot, and it should be beginner-friendly.

Furthermore, I would think a concrete example is much better than using variables in the text (like `${custom_dir}` which is rather cryptic). Interestingly, many users of this project are technical writers, not developers. Also, as I can read from the results of the survey (which I will release some time soon), people have trouble customizing, as they seem to lack concrete examples. Could we take the Bootstrap icons (which I find to be an excellent use case, as it's a ready-to-use icon set) as an example from the start?

diba1013 commented:
I have rewritten the tutorial from the ground up to be more detailed and to remove the navigation links.

For now, I have removed the integration part with mkdocs-material-extensions. Although it is now possible (thanks to the fast and awesome work of @facelessuser), it [requires a bit more setup right now](https://github.com/diba1013/mkdocs-material-playground/compare/feature/icons-in-markdown) (mostly upgrading to the latest version and adding a stylesheet). I will prepare a corresponding PR (once this is merged) where we can discuss the necessary adaptions.

facelessuser commented:
@diba1013, stylesheet should not be necessary. You're own custom icons should be styled the same as  all the other icons that material includes...unless your SVG have additional things that must be overridden.

diba1013 commented:
> stylesheet should not be necessary

Yes, I think I was still using `mkdocs-material==5.0.0` which does not style the icons properly (i.e. original size). I will confirm this prior to creating the PR.

squidfunk commented:
Great! I'll rebase and merge it into the docs branch and we can improve from there on 👍
